src/tests/CharityVaultFactory.t.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import {MockERC20} from "solmate/tests/utils/MockERC20.sol";
+
+import {DSTestPlus} from "./utils/DSTestPlus.sol";
+import "../CharityVaultFactory.sol";
+import "../CharityVault.sol";
+
+contract CharityVaultFactoryTest is DSTestPlus {
+    CharityVaultFactory factory;
+    MockERC20 underlying;
+
+    function setUp() public {
+        underlying = new MockERC20("Mock Token", "TKN", 18);
+        factory = new CharityVaultFactory();
+    }
+
+
+    // The CharityVaultFactory should not be able to find a deployed vault for anything initially
+    function test_basic_charity_vault_not_deployed(address fuzzed_addr) public {
+        assertTrue(
+            !factory.isVaultDeployed(CharityVault(payable(fuzzed_addr)))
+        );
+    }
+
+    /// @notice we can deploy a CharityVault
+    /// @dev uses the same testing pattern as VaultFactory for consistency
+    function test_able_to_deploy_charity_vault_from_factory(address fuzzed_addr, uint256 feePercent) public {
+      // Validate fuzzed fee percent
+      if (feePercent > type(uint256).max / 1e36) return;
```

Comment:
hm doesn't look like this bound is enforced in the contracts anywhere? if u don't want to accept fees over a certain % mayb validate in the charity vault constructor